packages/shared/src/models/IContainerConnectionInfo.ts

```diff
@@ -0,0 +1,43 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export type ContainerConnectionInfo =
+  | RunningContainerConnection
+  | NotEnoughResourcesContainerConnection
+  | NoContainerConnection;
+
+export type ContainerConnectionInfoStatus = 'running' | 'no-machine' | 'not-enough-resources';
```

Comment:
`not-enough-resources` seems a bit rash, maybe `low-resources` ?

Reply:
For specific models, like embedding which requires almost no ram, a few GB of RAM are enough